File: Frontend/src/types/Product.ts
// Product interface matching backend response
export interface Product {
  name: string;
  popularityScore: number;
  weight: number;
  price: number;
  images: {
    yellow: string;
    rose: string;
    white: string;
  };
  popularityScoreOutOfFive: number;
}

// Color options
export const COLOR_OPTIONS = {
  'yellow': '#E6CA97',
  'white': '#D9D9D9',
  'rose': '#E1A4A9'
} as const;

export type ColorOption = keyof typeof COLOR_OPTIONS;

// API Filter Parameters
export interface ApiFilterParams {
  minPrice?: number;
  maxPrice?: number;
  minPopularity?: number;
  maxPopularity?: number;
}

File: Backend/src/main/java/com/mrvalevictorian/backend/models/Product.java
package com.mrvalevictorian.backend.models;

import lombok.*;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class Product {
    private String name;
    private double popularityScore;
    private double weight;
    private double price;
    private Images images;
    private double popularityScoreOutOfFive;
}
